internal/ethapi/simulate.go

```diff
@@ -200,7 +200,7 @@ func (sim *simulator) processBlock(ctx context.Context, block *simBlock, header,
 		if err := ctx.Err(); err != nil {
 			return nil, nil, err
 		}
-		if err := sim.sanitizeCall(&call, sim.state, header, blockContext, &gasUsed); err != nil {
+		if err := sim.sanitizeCall(&call, tracingStateDB, header, blockContext, &gasUsed); err != nil {
```

Comment:
This is a change in semantics. I think this is wrong, the `sanitizeCall` is only used for setting nonce if unset: 

```
		nonce := state.GetNonce(call.from())
		call.Nonce = (*hexutil.Uint64)(&nonce)
```
I don't think we should use the hooked state for that. 

```suggestion
		if err := sim.sanitizeCall(&call, sim.state, header, blockContext, &gasUsed); err != nil {
```

Reply:
Hm, actually, `sanitizeCall` is only called from one location (here), you could just do the nonce-check outside, and skip passing the state along